website/docs/cdktf/debugging.mdx

```diff
@@ -0,0 +1,59 @@
+---
+page_title: Debugging
+description: >-
+  Learn how to troubleshoot problems in your CDKTF application.
+---
+
+# Debugging
+
+## Increasing the verbosity
+
+To get more information about what CDKTF is doing and what problems have occured by setting the `CDKTF_LOG_LEVEL` environment variable to `DEBUG`. If you want to share the logs you can set `CDKTF_LOG_FILE_DIRECTORY` to a file path, a `cdktf.log` file with all logs will be written.
+
+## Troubleshooting
+
+To understand where the problem is coming from it's a good first step to see if it's a synth-time issue or an execution-time issue.
+If `cdktf synth` fails you have a synth-time issue, otherwise it's an execution-time issue.
+
+### Synth-time issues
+
+The most important thing is to understand which construct is causing the problem. The error message thrown might hint you towards it, if not you will need to search for it. If you have a last version that successfully synthesized you can search through the new changes, otherwise you need to search through all changes. To search it you need to repeatedly run `cdktf synth` and comment parts out until you find the error. The synthesized code does not need to make any logical sense since we don't plan to apply it.
+
+Once you find the construct causing the problem you need to try to isolate the problem. If you use variables in your constructs configuration, try to temporarily replace them with static values to ensure the problem is on the construct itself. If the problem persists try looking at the type to identfy missing or mistyped fields. Looking at the resource in the terraform provider might help to understand the structure better as well.
+
+If this does not help, please create an issue in the [hashicorp/terraform-cdk](https://github.com/hashicorp/terraform-cdk/issues/new?assignees=&labels=bug%2C+new&template=bug-report.md&title=) repository.
+
+### Execution-time issues
+
+Execution-time issues can arise from misconfigured synthesized code or from the provider itself.
+As a first step you can check if the synthesized code looks like you would expect it. You can check through the Terraform documentation for the resource to see if there are misconfigured values. You can also search for the resource name, terraform and the error to find issues Terraform users had with the resource or look at the providers github repository for issues similar to yours. Keep in mind that in CDKTF renames resources and fields in most languages to be camel case, whereas terraform uses snake case.
```

Comment:
Users unfamiliar with HCL might not know what to expect

Reply:
Maybe @laurapacilio can help? Else I'd be fine with leaving this as is, as this is quite an advanced topic already.